src/main/java/com/management/service/CarService.java

```diff
@@ -0,0 +1,79 @@
+package com.management.service;
+
+import com.management.model.CarModel;
+import com.management.repository.CarRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.regex.Pattern;
+
+@Service
+public class CarService {
+
+	private final CarRepository carRepository;
+
+	@Autowired
+	public CarService(CarRepository carRepository) {
+		this.carRepository = carRepository;
+	}
+
+	public Optional<CarModel> saveCar(CarModel carModel) {
+		// Verifica se a placa é null antes de validar
+		if (carModel.getLicensePlate() != null) {
+			validateLicensePlate(carModel.getLicensePlate());
+		} else {
+			// Considerar lançar uma exceção ou tratar o caso de placa null conforme a regra
+			// de negócio
+			throw new IllegalArgumentException("A placa do veículo não pode ser nula.");
+		}
+
+		Optional<CarModel> existingCar = carRepository.findByBrandAndModelAndYearAndLicensePlate(carModel.getBrand(),
+				carModel.getModel(), carModel.getYear(), carModel.getLicensePlate());
+		if (existingCar.isPresent()) {
+			// Carro já existe, então retornamos um Optional vazio para indicar conflito
+			return Optional.empty();
+		} else {
+			// Carro não existe, então salvamos e retornamos em um Optional
+			return Optional.of(carRepository.save(carModel));
+		}
```

Comment:
Que tal se ao inves de retornar `return Optional.empty();` ele subir uma exception, então no controller ao tratar a esception em questão ele devolve 409 conflict?

Reply:
Done, change for exception create with new packge com.management.exception for these situation